### PR TITLE
132 display created projects

### PR DESCRIPTION
**Pull request by muratcan23**

### CODE REVIEW

File: src/components/Navbar/Components/ProjectCard.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          alignItems="center"
+          h="32px"
+          borderRadius="5px"
+          p="1px"
+          _hover={{ backgroundColor: "#eeeeee", cursor: "pointer" }}
+        >
+          <ColorDotIcon color={project.color} />
+          <Text fontSize="16px">{project.name}</Text>
+          <Flex ml="auto">...</Flex>
+        </Flex>
```
Comment (kod-man) [inline src/components/Navbar/Components/ProjectCard.tsx:53]: This part can be called ProjectItemCard. You can create new component

**Review by kod-man — CHANGES_REQUESTED** [review]

File: src/components/Navbar/Components/DrawerComponent.tsx [hunk 2]
```diff
@@ -92,7 +94,7 @@ const DrawerComponent: FC<DrawerProps> = ({ isOpen, onClose }) => {
           </Flex>
         </Flex>
 
-        {isProjectListOpen && <ProjectCardComponent name="deneme" color="gray" />}
+        {isProjectListOpen && <ProjectCard name="deneme" color="gray" />}
```
Comment (kod-man) [inline src/components/Navbar/Components/DrawerComponent.tsx:97]: what is the purpose of having a name and color props?

File: src/components/Navbar/Components/ProjectCard.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  name: string;
+};
+
+type Project = {
+  id: number;
+  name: string;
+  color: string;
+};
+
+function ProjectCard({ name, color }: ProjectCardProps) {
```
Comment (kod-man) [inline src/components/Navbar/Components/ProjectCard.tsx:17]: Do you use name and color props?

File: src/components/Navbar/Components/ProjectCard.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  const [projects, setProjects] = useState<Project[]>([]);
+
+  useEffect(() => {
+    const userId = sessionStorage.getItem("userId");
+    if (!userId) return;
+
+    Axios.get(`${API.getProject}/${userId}`)
+      .then((response) => {
+        console.log(response.data);
```
Comment (kod-man) [inline src/components/Navbar/Components/ProjectCard.tsx:26]: remove console

File: src/components/Navbar/Components/ProjectCard.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        setProjects(response.data);
+      })
+      .catch((err) => {
+        console.log(err);
+        setProjects([]);
```
Comment (kod-man) [inline src/components/Navbar/Components/ProjectCard.tsx:31]: Why are you setting projects []? What about if we already had 5 projects and got an error while adding 6th? Will you remove the existing ones?

File: src/components/Navbar/Components/ProjectCard.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      });
+  }, []);
+
+  return (
+    <>
+      {projects.map((project) => (
```
Comment (kod-man) [inline src/components/Navbar/Components/ProjectCard.tsx:37]: If you use map, you better use key props as well

File: src/components/Navbar/Components/ProjectItemCard.tsx [hunk 7]
```diff
@@ -1,14 +1,16 @@
...
+  id: number;
   name: string;
+  color: string;
 };
 
-function ProjectCardComponent({ name, color }: ProjectCardComponentProps) {
+const ProjectItemCard = ({ id, name, color }: ProjectItemCardProps) => {
   return (
     <Flex
+      key={id}
```
Comment (kod-man) [inline src/components/Navbar/Components/ProjectItemCard.tsx:13]: we don't need key here. We need key on the upper level

**Review by kod-man — CHANGES_REQUESTED** [review]

File: src/components/Navbar/Components/ProjectCard.tsx (hunk 5 above)
Comment (muratcan23) [reply src/components/Navbar/Components/ProjectCard.tsx:31]: Sifu did you mean this:  ` setProjects((prevProjects) => [...prevProjects, newProject])`

File: src/components/Navbar/Components/DrawerComponent.tsx (hunk 2 above)
Comment (muratcan23) [reply src/components/Navbar/Components/DrawerComponent.tsx:97]: unnecessary props will be erased.

File: src/components/Navbar/Components/ProjectCard.tsx (hunk 3 above)
Comment (muratcan23) [reply src/components/Navbar/Components/ProjectCard.tsx:17]: unnecessary props will be erased.

**Pull request merged**